trie-db/src/lookup.rs

```diff
@@ -22,20 +22,22 @@ use crate::{
 	Bytes, CError, CachedValue, DBValue, MerkleValue, Query, RecordedForKey, Result, TrieAccess,
 	TrieCache, TrieError, TrieHash, TrieLayout, TrieRecorder,
 };
-use hash_db::{HashDBRef, Hasher, Prefix};
+use hash_db::{HashDB, Hasher, Prefix};
 
 /// Trie lookup helper object.
 pub struct Lookup<'a, 'cache, L: TrieLayout, Q: Query<L::Hash>> {
 	/// database to query from.
-	pub db: &'a dyn HashDBRef<L::Hash, DBValue>,
+	pub db: &'a dyn HashDB<L::Hash, DBValue, L::Location>,
 	/// Query object to record nodes and transform data.
 	pub query: Q,
 	/// Hash to start at
 	pub hash: TrieHash<L>,
+	/// Optionally location to start at.
+	pub location: L::Location,
```

Comment:
From arkpar/new-trie I did add location to lookup which remove the need to pass it in parameter in lookup api.